R/utils.R

```diff
@@ -127,6 +127,41 @@ parse_answer = function(api_answer, multiple = FALSE, key = NULL) {
   }
 }
 
+#' Parse publisher list
+#'
+#' @param api_answer
+#'
+#' @return a parsed data.frame of publisher
+parse_publisher = function(api_answer) {
+  if (http_error(api_answer)) {
+    stop("The API endpoint could not be reached. Please try again later.")
+  }
+
+  xml_source = content(api_answer, encoding = "ISO-8859-1")
+
+  apicontrol = xml_text(xml_find_all(xml_source, "//apicontrol"))
+
+  if (apicontrol == "invalidkey") {
+    stop("The provided API key is invalid. ",
+         "You can register for a free API at ",
+         "http://www.sherpa.ac.uk/romeo/apiregistry.php")
+  }
+
+  publisher   = xml_text(xml_find_all(xml_source, "//name"))
+  romeocolour = xml_text(xml_find_all(xml_source, "//romeocolour"))
+  preprint    = xml_text(xml_find_all(xml_source, "//prearchiving"))
+  postprint   = xml_text(xml_find_all(xml_source, "//postarchiving"))
+  pdf         = xml_text(xml_find_all(xml_source, "//pdfarchiving"))
+  pdf         = ifelse(pdf == "unknown", NA_character_, pdf)
+
+  data.frame(publisher   = publisher,
```

Comment:
I'm not convinced that `publisher` is the most relevant field to return. Or at least, not alone:

![screenshot from 2018-11-27 17-21-45](https://user-images.githubusercontent.com/10783929/49095560-0722a900-f269-11e8-939b-3e2ccdccfde8.png)
![screenshot from 2018-11-27 17-21-22](https://user-images.githubusercontent.com/10783929/49095561-0722a900-f269-11e8-8c73-3704525275b2.png)

Reply:
I tried returning the `alias` field but it does not seem better either... Sometimes it's clearer otherwise irrelevant
Relevant example: http://www.sherpa.ac.uk/romeo/api29.php?id=1532
Irrelevant example: http://www.sherpa.ac.uk/romeo/api29.php?id=2438
What do you think?

Reply:
I added `alias` and `romeoid` fields in hope that it makes the policies more relevant.